src/addrman.h

```diff
@@ -142,6 +165,15 @@ class AddrMan
     void SetServices(const CService& addr, ServiceFlags nServices);
 
     const std::vector<bool>& GetAsmap() const;
+
+    /** Test-only function
+     * Find the address record in AddrMan and return information about its
+     * position.
+     * @param[in] addr       The address record to look up.
+     * @return               Information about the address record in AddrMan
+     *                       nullptr if address is not found
+     */
+    std::optional<AddressPosition> FindAddressEntry(const CAddress& addr);
```

Comment:
nit: if this is a test-only function, would not it be better to implement it in a derived class in the test file (such as the deleted class `AddrManTest`)?

Reply:
I wanted to avoid reaching into the AddrMan internals in the unit tests as it is done currently. Also, I think that this function can be of use in the fuzz tests as well, which also still directly access internals in some spots.
One alternative possibility I thought about was adding `FindAddressEntry()` only to `addrman_impl.h` instead of `addrman.h` and keep on working with derived classes in the test accessing it that way. If people would prefer, I could switch to that.
Note that a very limited set of test-only functions in the public interface also exist in comparable places such as `txrequest.h`